### DISCUSSION TITLE
Project Template Integration

### DISCUSSION THREAD

**Discussion by Spokeek** [vrc-get/vrc-get#1512] [Ideas]
I wanted to open this discussion to suggest this feature we discussed a while ago in this issue https://github.com/vrc-get/vrc-get/issues/852

Today VCC can create project through templates, mostly by duplicating a project.
I believe that's a good feature to add to ALCOM but it can be thought further and that features can be added to it.

I was thinking on a way for people to add templates like we do for vcc repositories, and having a way to sync those templates se same way that we sync github forks, or maybe even through git releases (i don't know yet).

What are you thoughts on that and what kind of features would you see ?

This is IMO a good way for ALCOM to differenciate from VCC by providing more while staying compliant with standards.

**Comment by anatawa12** [comment]
> Today VCC can create project through templates, mostly by duplicating a project.
> I believe that's a good feature to add to ALCOM

ALCOM already have that now including custom templates support.

I'm for for distribute / download project templates with something like repository but I have concerns withfuture VCC compatibility.

there is one issue on vcc side for distributing templetes (153) and if it become available for community, we may have two formats for project templetes. This should be avoided horrible.

I understood #852 as applying changes of templetes to already created project but it seem I misread your feature request. Sorry.

I don't like to use git for templete distribution since it's unlikely for general Users to have git installed. I think it's good to have JSON file for general info and zip or tar.gz (or tar.xz tar.zstd for better compression) for contents of templete, for now.

**Reply by Spokeek** [reply]
I agree that the use of git isn't ideal, that's what i used on my side but that's not the proposal i wanted to make.
Using an archive with versioning with a json config file seem to be a good way to go.

The issue you mention is https://github.com/vrchat-community/creator-companion/issues/153 if i'm not wrong and right now they download a list of templates from a remote source, but i think this is for convenience, but the "spec" seem clear to me.

We would have a list of folders, each folder is a clean unity project with a package.json file following the next structure

```
{
	"name" : "com.vrchat.template.base",
	"displayName" : "Base VRChat Project",
	"description" : "",
	"version" : "1.0.2",
	"category" : "ProjectTemplate",
	"dependencies" : {}
}
```

We can already use this as a base to work.

A good first feature i would see is the ability to create custom template from within the GUI as we do for user defined packages.
